### PR TITLE
Implement scope-local C pointer.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: INTEROP.md [hunk 1]
```diff
@@ -337,6 +337,22 @@ manually:
 
 In all cases the C string is supposed to be encoded as UTF-8.
 
+### Scope-local pointers ###
+
+It is possible to create scope-stable pointer of C representation of `CValues<T>`
+instance using `MemScope.ptr` extension property. It allows to use APIs which requires C
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: It's better to refer this property as "`CValues<T>.ptr` extension property available under `memScoped { ... }`".

Comment (olonho) [reply INTEROP.md]: Done

File: samples/androidNativeActivity/src/main/kotlin/renderer.kt [hunk 2]
```diff
@@ -306,10 +306,10 @@ class Renderer(val parentArena: NativePlacement, val nativeActivity: ANativeActi
         }
 
         glFrontFace(GL_CW)
-        glVertexPointer(4, GL_FLOAT, 0, vertices.toFloatArray().toCValues().getPointer(this))
-        glTexCoordPointer(2, GL_FLOAT, 0, texCoords.toFloatArray().toCValues().getPointer(this))
-        glNormalPointer(GL_FLOAT, 0, normals.toFloatArray().toCValues().getPointer(this))
-        glDrawElements(GL_TRIANGLES, triangles.size, GL_UNSIGNED_BYTE, triangles.toByteArray().toCValues().getPointer(this))
+        glVertexPointer(4, GL_FLOAT, 0, vertices.toFloatArray().toCValues().ptr)
```
Comment (abreslav) [inline samples/androidNativeActivity/src/main/kotlin/renderer.kt]: Looks like `toCValues().ptr` is a rather common operation. Do you think we could add a shortcut for it?

File: samples/uikit/src/main/kotlin/main.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@ import platform.UIKit.*
 fun main(args: Array<String>) {
     memScoped {
         val argc = args.size + 1
-        val argv = (arrayOf("konan") + args).map { it.cstr.getPointer(memScope) }.toCValues()
+        val argv = (arrayOf("konan") + args).map { it.cstr.ptr }.toCValues()
```
Comment (abreslav) [inline samples/uikit/src/main/kotlin/main.kt]: What's the difference between `getPointer(memScope)` and `getPointer(this)`?

Comment (olonho) [reply samples/uikit/src/main/kotlin/main.kt]: Those are synonymous: https://github.com/JetBrains/kotlin-native/blob/0a911ed5bbf5413f54facc0b37961cfc3fae0310/Interop/Runtime/src/main/kotlin/kotlinx/cinterop/Utils.kt#L413

File: samples/androidNativeActivity/src/main/kotlin/renderer.kt (hunk 2 above)
Comment (olonho) [reply samples/androidNativeActivity/src/main/kotlin/renderer.kt]: Problem is that we have multiple toCValues() methods, so we must provides multiple shortcuts then, and semantics would be pretty magical.

**Pull request merged**